CorsixTH/Lua/dialogs/fullscreen/progress_report.lua

```diff
@@ -49,52 +48,35 @@ function UIProgressReport:UIProgressReport(ui)
 
   -- Selected hospital number
   self.selected = 1
+  local hospital = world.hospitals[self.selected]
 
+  -- Collect which criteria to show here, draw columns in draw
   -- Add the icons for the criteria
   local x = 263
-  local world_goals = world.goals
-  for _, tab in ipairs(world_goals) do
-    local crit_name = world.level_criteria[tab.criterion].name
-    local res_value = world_goals[crit_name].win_value
-    world_goals[crit_name].visible = true
-    -- Special case for money, subtract loans
-    local cur_value = hospital[crit_name]
-    if crit_name == "balance" then
-      cur_value = cur_value - hospital.loan
-    end
-    if world_goals[crit_name].lose_value then
-      world_goals[crit_name].red = false
-
-      if cur_value < world_goals[crit_name].boundary then
-        world_goals[crit_name].red = true
-        res_value = world_goals[crit_name].lose_value
-        -- TODO: Make the ugly workaround for the special case "percentage_killed" better
-        if crit_name:find("killed") then
-          res_value = nil
-          world_goals[crit_name].visible = false
-        end
-      elseif not world_goals[crit_name].win_value then
-        world_goals[crit_name].visible = false
+  if not world.free_build_mode then
+    local crit_data = world.endconditions:generateReportTable(hospital)
+    for i, crit_table in ipairs(crit_data) do
+      crit_table.visible = true
+      local crit_name = crit_table.name
+      local res_value = crit_table.win_value
+      local cur_value = world.endconditions:getAttribute(hospital, crit_name)
+      if crit_table.lose_value then
+        crit_table.red = true
+        res_value = crit_table.lose_value
       end
-    end
-    -- Only five criteria can be there at once.
-    if crit_name:find("killed") and world.winning_goal_count > 5 then
-      res_value = nil
-      world_goals[crit_name].visible = false
-    end
-    if res_value then
       -- FIXME: res_value and cure_value are depersisted as floating points, using
       -- string.format("%.0f", x) is not suitable due to %d (num) param in _S string
       local tooltip
-      if world.level_criteria[tab.criterion].formats == 2 then
+      if crit_table.formats == 2 then
         tooltip = _S.tooltip.status[crit_name]:format(math.floor(res_value), math.floor(cur_value))
       else
         tooltip = _S.tooltip.status[crit_name]:format(math.floor(res_value))
       end
```

Comment:
Wouldn't `formats` be better named as `number_of_values` ?

Reply:
I have plans for other formats that are two values,
```lua
       elseif crit_table.formats == 3 then
         tooltip = tooltip:format(math.floor(res_value) / 100, math.floor(cur_value) / 100)
```